user-page/user_homepage/script.js

```diff
@@ -146,30 +132,37 @@ function appendHistoryLog(data, divId) {
     };
 };
 
+async function getHistoryLogs() {
+    const endpoint = "historyLogsData.json";
+    const key = "historyLogsData"
 
-//get data from historyLogsData.json file
-$.ajax({
-    url: "historyLogsData.json",
-    success: function(data) {
-
-        // get string from json
-        // var fetchedData = JSON.stringify(data);
-
-        console.log(data);
-
-        //is this really a good idea? how else can i store it?
-        localStorage.setItem("dataFromJson", data);
-    }
-});
-
+    return $.ajax({
+        url: endpoint,
+        success: function(data) {
 
+            //is this really a good idea? how else can i store it?
+            // data = JSON.parse(JSON.stringify(data))
+            sessionStorage.setItem("dataFromJson", data);
+            return data
+        }
+    });
+};
 
 
 
-// console.log(localStorage.getItem("dataFromJson"));
-//data = localStorage.getItem("dataFromJson");
+getHistoryLogs().then(function(gottenData) {
+        console.log(gottenData);
+        data = gottenData;
+    })
+    //after get request is succes, append data to page
+    .then(function() {
+        for (let index = 0; index < 10; index++) {
+            appendHistoryLog(data[index], "session" + index);
+        }
+    })
+    .then(function() {
+        let storedData = sessionStorage.getItem("dataFromJson");
+        storedData = JSON.parse(JSON.stringify(storedData));
+        console.log(storedData);
```

Comment:
on line 160 you call the function appendHistoryLog but you didn't convert the data to JSON. You do this at line 165. 
Add line before 158 `data_json = JSON.parse(JSON.stringify(data));` and replace the data on line 160 to data_json